### PR TITLE
MAISTRA-280: Put Grafana/Prometheus/Jaeger consoles behind authentication

### PR DESCRIPTION
**Pull request by jwendell**
Using OpenShift OAuth proxy.

### CODE REVIEW

File: helm/istio/charts/tracing/templates/deployment-jaeger.yaml [hunk 1]
```diff
@@ -113,6 +113,43 @@ spec:
 {{- end }}
       serviceAccountName: jaeger
       containers:
+      # OAuth proxy
```
Comment (rcernich) [inline helm/istio/charts/tracing/templates/deployment-jaeger.yaml:116]: As we're moving to the Jaeger operator, does the operator support this configuration?

**Review by rcernich — COMMENTED** [review]
Overall, this looks good.  One question, as we're migrating to using the Jaeger operator, would it be better to hold off on the Jaeger related changes and make sure they're incorporated into that migration?

**Comment by jotak** [comment]
We'll have to test Kiali against this PR. Any already known issue (in Kiali) about that?

**Comment by jwendell** [comment]
@jotak Not that I know, from this PR itself. I mean, with it I can open Jaeger, Prometheus and Grafana console UI's just fine. When I open Kiali UI I see a warning about it not connecting to Grafana, but this I'll let you guys investigate (hint: the Grafana URL scheme has changed to HTTPS)

**Comment by jotak** [comment]
@jwendell can you elaborate how it is supposed to work, and which permissions I should grant to get access to Grafana for instance?

Because I'm surprised I get "403: unauthorized " using an admin access token (I have cluster-role admin access).

Also, when I try to connect from the browser, I'm redirected to openshift login page (so far so good) but when login i'm having "500 internal error" displayed. With following logs on grafana-proxy container:

```
2019/06/06 16:21:25 provider.go:553: Performing OAuth discovery against https://172.30.0.1/.well-known/oauth-authorization-server 
2019/06/06 16:21:25 provider.go:593: 200 GET https://172.30.0.1/.well-known/oauth-authorization-server  {
  "issuer": "https://127.0.0.1:8443 ",
  "authorization_endpoint": "https://127.0.0.1:8443/oauth/authorize ",
  "token_endpoint": "https://127.0.0.1:8443/oauth/token ",
  "scopes_supported": [
    "user:check-access",
    "user:full",
    "user:info",
    "user:list-projects",
    "user:list-scoped-projects"
  ],
  "response_types_supported": [
    "code",
    "token"
  ],
  "grant_types_supported": [
    "authorization_code",
    "implicit"
  ],
  "code_challenge_methods_supported": [
    "plain",
    "S256"
  ]
}
2019/06/06 16:21:26 oauthproxy.go:646: error redeeming code (client:172.17.0.1:42592): Post https://127.0.0.1:8443/oauth/token:  dial tcp 127.0.0.1:8443: getsockopt: connection refused
2019/06/06 16:21:26 oauthproxy.go:439: ErrorPage 500 Internal Error Internal Error
2019/06/06 16:21:32 server.go:2753: http: TLS handshake error from 172.17.0.1:42704: EOF

```

Not sure if it's related to the 403 I get above, or if it's two different things.

**Comment by xeviknal** [comment]
@jotak I only tried to access via browser and I had no problems. I got redirected to the openshift login page (OAuth) and with classic admin/admin I could access to the grafana dashboard.

```
2019/06/07 10:47:09 provider.go:553: Performing OAuth discovery against https://172.30.0.1/.well-known/oauth-authorization-server 
2019/06/07 10:47:09 provider.go:593: 200 GET https://172.30.0.1/.well-known/oauth-authorization-server  {
  "issuer": "https://192.168.42.50:8443 ",
  "authorization_endpoint": "https://192.168.42.50:8443/oauth/authorize ",
  "token_endpoint": "https://192.168.42.50:8443/oauth/token ",
  "scopes_supported": [
    "user:check-access",
    "user:full",
    "user:info",
    "user:list-projects",
    "user:list-scoped-projects"
  ],
  "response_types_supported": [
    "code",
    "token"
  ],
  "grant_types_supported": [
    "authorization_code",
    "implicit"
  ],
  "code_challenge_methods_supported": [
    "plain",
    "S256"
  ]
}
2019/06/07 10:47:10 provider.go:593: 200 GET https://172.30.0.1/apis/user.openshift.io/v1/users/~  {"kind":"User","apiVersion":"user.openshift.io/v1","metadata":{"name":"admin","selfLink":"/apis/user.openshift.io/v1/users/admin","uid":"e00f6482-890b-11e9-8140-52540030ff0a","resourceVersion":"2549","creationTimestamp":"2019-06-07T10:06:10Z"},"fullName":"admin","identities":["anypassword:admin"],"groups":["system:authenticated","system:authenticated:oauth"]}
2019/06/07 10:47:10 provider.go:593: 201 POST https://172.30.0.1/apis/authorization.openshift.io/v1/subjectaccessreviews  {"kind":"SubjectAccessReviewResponse","apiVersion":"authorization.openshift.io/v1","namespace":"istio-system","allowed":true,"reason":"RBAC: allowed by ClusterRoleBinding \"cluster-admin-1\" of ClusterRole \"cluster-admin\" to User \"admin\""}
2019/06/07 10:47:10 oauthproxy.go:676: 172.17.0.1:40492 authentication complete Session{admin@cluster.local token:true}
```

Same with prometheus.

**Comment by jotak** [comment]
FYI i don' have anymore the 500 internal error, but still cannot `curl` the APIs with bearer token. It's probably what @mwringe said in a mail about option `--openshift-delegate-urls`

**Comment by aljesusg** [comment]
from me side, with the last updates.

Jaeger is not available by eleasticsearch:

```bash
max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
```

and prometheus

```bash
level=info ts=2019-06-10T09:18:23.7376059Z caller=main.go:222 msg="Starting Prometheus" version="(version=2.3.1, branch=HEAD, revision=188ca45bd85ce843071e768d855722a9d9dabe03)"
level=info ts=2019-06-10T09:18:23.737649427Z caller=main.go:223 build_context="(go=go1.10.3, user=root@82ef94f1b8f7, date=20180619-15:56:22)"
level=info ts=2019-06-10T09:18:23.737672324Z caller=main.go:224 host_details="(Linux 4.20.6-200.fc29.x86_64 #1 SMP Thu Jan 31 15:50:43 UTC 2019 x86_64 prometheus-5c79f4f744-2j2ws (none))"
level=info ts=2019-06-10T09:18:23.737687365Z caller=main.go:225 fd_limits="(soft=1048576, hard=1048576)"
level=info ts=2019-06-10T09:18:23.738246735Z caller=main.go:514 msg="Starting TSDB ..."
level=info ts=2019-06-10T09:18:23.738261486Z caller=web.go:415 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2019-06-10T09:18:23.738299721Z caller=main.go:402 msg="Stopping scrape discovery manager..."
level=info ts=2019-06-10T09:18:23.738321872Z caller=main.go:416 msg="Stopping notify discovery manager..."
level=info ts=2019-06-10T09:18:23.73834144Z caller=main.go:438 msg="Stopping scrape manager..."
level=info ts=2019-06-10T09:18:23.738362672Z caller=manager.go:464 component="rule manager" msg="Stopping rule manager..."
level=info ts=2019-06-10T09:18:23.738372158Z caller=manager.go:470 component="rule manager" msg="Rule manager stopped"
level=info ts=2019-06-10T09:18:23.738382922Z caller=notifier.go:512 component=notifier msg="Stopping notification manager..."
level=info ts=2019-06-10T09:18:23.738395683Z caller=main.go:588 msg="Notifier manager stopped"
level=info ts=2019-06-10T09:18:23.738362645Z caller=main.go:432 msg="Scrape manager stopped"
level=info ts=2019-06-10T09:18:23.738410395Z caller=main.go:398 msg="Scrape discovery manager stopped"
level=info ts=2019-06-10T09:18:23.738419006Z caller=main.go:412 msg="Notify discovery manager stopped"
level=error ts=2019-06-10T09:18:23.738708037Z caller=main.go:597 err="Opening storage failed mkdir data/: permission denied"
level=info ts=2019-06-10T09:18:23.738734618Z caller=main.go:599 msg="See you next time!"
```

I can't reach grafana or jaeger by browser. @xeviknal @jotak Do the latest changes work with your environments?

**Comment by mwringe** [comment]
Is this configurable? Can we turn on or off the oauth proxy for each of the components?

**Comment by jwendell** [comment]
@mwringe no, not in this PR at least. Is this a requirement?
@knrc ^

**Comment by jwendell** [comment]
@mwringe Just talked to Kev and he confirmed there should not be an option to not use OAuth (at least for GA).

**Pull request merged**